# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/locale/LocaleService.cpp
# Commit: 6450088b6b73
# Full Hash: 6450088b6b73ffc17c79cd7097b1bfe30d00e207
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-09-14 21:29:41
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1656261 - Back out all recent StartupCache work r=RyanVM
#   
#   This backs out all work from bug 1627075 as well as all of its
#   descendents. There were a few conflicts when backing this out but
#   overall it was pretty clean, so I would say it's a fairly mild
# ==============================================================================

diff -r b0fabcd60430 -r 6450088b6b73 intl/locale/LocaleService.cpp
--- a/intl/locale/LocaleService.cpp	Mon Sep 14 17:32:40 2020 +0000
+++ b/intl/locale/LocaleService.cpp	Mon Sep 14 17:00:53 2020 +0000
@@ -295,26 +295,13 @@
 
 static bool GetGREFileContents(const char* aFilePath, nsCString* aOutString) {
   // Look for the requested file in omnijar.
-  RefPtr<CacheAwareZipReader> zip = Omnijar::GetReader(Omnijar::GRE);
+  RefPtr<nsZipArchive> zip = Omnijar::GetReader(Omnijar::GRE);
   if (zip) {
-    const auto item = zip->GetItem(aFilePath);
+    nsZipItemPtr<char> item(zip, aFilePath);
     if (!item) {
       return false;
     }
-
-    MOZ_ASSERT(item->RealSize());
-
-    auto buf = MakeUnique<uint8_t[]>(item->RealSize());
-    CacheAwareZipCursor cursor(item, zip, buf.get(), item->RealSize());
-
-    uint32_t count;
-    uint8_t* data = cursor.Read(&count);
-
-    if (count != item->RealSize()) {
-      return false;
-    }
-
-    aOutString->Assign(reinterpret_cast<const char*>(data), count);
+    aOutString->Assign(item.Buffer(), item.Length());
     return true;
   }
 